py/redrock/targets.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, wave, flux, ivar, R, Rcsr=None):
         self.R = R
         self._Rcsr = Rcsr
         self._mpshared = False
-        self.wavehash = hash((len(wave), wave[0], wave[1], wave[-2], wave[-1]))
+        self.wavehash = hash((len(wave), wave[0], wave[1], wave[-2], wave[-1], R.data.shape[0]))
```

Comment:
From this code, if different spectra of the same object have different numbers of diagonal elements in the resolution matrix then they will have different `wavehash` values and thus will not be coadded.  Is that the desired behavior?  That could be ok since they will still be fit as part of the same target for redshift fitting.  If you wanted, it would still be possible to coadd these using fancier indexing logic in `Target.compute_coadd`, but that may not be necessary.

Reply:
It is the behavior that I want. It is ok if they are not coadded, as long as they are all used to fit the target redshift.